File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationSameWidthAs.java
package com.github.florent37.expectanim.core.scale;

import android.view.View;

/**
 * Created by florentchampigny on 20/02/2017.
 */
public class ScaleAnimExpectationSameWidthAs extends ScaleAnimExpectationViewDependant {

    public ScaleAnimExpectationSameWidthAs(View otherView, Integer gravityHorizontal, Integer gravityVertical) {
        super(otherView, gravityHorizontal, gravityVertical);
    }

    @Override
    public Float getCalculatedValueScaleX(View viewToMove) {
        final int viewToMoveWidth = viewToMove.getWidth();

        final float otherViewWidth = viewCalculator.finalWidthOfView(otherView);

        if (otherViewWidth == 0 || viewToMoveWidth == 0f) {
            return 0f;
        }
        return otherViewWidth / viewToMoveWidth;
    }

    @Override
    public Float getCalculatedValueScaleY(View viewToMove) {
        if(keepRatio){
            return getCalculatedValueScaleX(viewToMove);
        }
        return null;
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/scale/ScaleAnimExpectationSameWidthAs.kt
package com.github.florent37.kotlin.pleaseanimate.core.scale

import android.view.View

/**
 * Created by florentchampigny on 20/02/2017.
 */
class ScaleAnimExpectationSameWidthAs(otherView: View, gravityHorizontal: Int?, gravityVertical: Int?) : ScaleAnimExpectationViewDependant(otherView, gravityHorizontal, gravityVertical) {

    override fun getCalculatedValueScaleX(viewToMove: View): Float? {
        val viewToMoveWidth = viewToMove.width

        val otherViewWidth = viewCalculator!!.finalWidthOfView(otherView)

        return if (otherViewWidth == 0f || viewToMoveWidth.toFloat() == 0f) {
            0f
        } else otherViewWidth / viewToMoveWidth
    }

    override fun getCalculatedValueScaleY(viewToMove: View): Float? {
        return if (keepRatio) {
            getCalculatedValueScaleX(viewToMove)
        } else null
    }
}
